script.js

```diff
@@ -1,12 +1,70 @@
+let allEpisodes;
+
 //You can edit ALL of the code here
 function setup() {
-  const allEpisodes = getAllEpisodes();
+  allEpisodes = getAllEpisodes();
   makePageForEpisodes(allEpisodes);
 }
 
+const search = document.getElementById("searchEpisode");
+const rootElem = document.getElementById("root");
+
+search.addEventListener("input",(element) => {
+  const value = element.target.value.toLowerCase();
+  //console.log(value)
+
+
+  const filterEpisodes = allEpisodes.filter((episode) =>
+       episode.name.toLowerCase().includes(value) ||
+        episode.summary.toLowerCase().includes(value)
+        );
+
+        console.log(filterEpisodes);
+
+
+        rootElem.innerHTML ="";
+        makePageForEpisodes(filterEpisodes); //////continue
+})
+
+
+
+
+
+
+
+
 function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
-}
+ 
+  //rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+
+  episodeList.forEach((episode) => {
+    // Add elements
+    const card = document.createElement("div");
+    card.classList.add("card");
+    const cardHeader = document.createElement("div");
+    cardHeader.classList.add("cardHeader");
+    const title = document.createElement("h3");
+    const season = document.createElement("h4");
+    const image = document.createElement("img");
+    const summary = document.createElement("p");
+    summary.classList.add("summary");
 
+    // Add content to your elements 
+    title.textContent = episode.name;
+    season.textContent = `S${episode.season
+      .toString()
+      .padStart(2, "0")}E${episode.number.toString().padStart(2, "0")}`;
```

Comment:
Clever, forgot about padStart I was like is that even a function and had to Google it 🤣